core/src/main/java/bio/terra/pearl/core/dao/participant/EnrolleeDao.java

```diff
@@ -117,18 +117,27 @@ public Optional<Enrollee> findByPreEnrollResponseId(UUID preEnrollResponseId) {
     public Enrollee loadForAdminView(Enrollee enrollee) {
         enrollee.getSurveyResponses().addAll(surveyResponseDao.findByEnrolleeIdWithAnswers(enrollee.getId()));
         enrollee.getConsentResponses().addAll(consentResponseDao.findByEnrolleeId(enrollee.getId()));
-        enrollee.setProfile(profileDao.loadWithMailingAddress(enrollee.getProfileId()).get());
-        enrollee.getParticipantTasks().addAll(participantTaskDao.findByEnrolleeId(enrollee.getId()));
         if (enrollee.getPreEnrollmentResponseId() != null) {
             enrollee.setPreEnrollmentResponse(preEnrollmentResponseDao.find(enrollee.getPreEnrollmentResponseId()).get());
         }
+        enrollee.getParticipantNotes().addAll(participantNoteDao.findByEnrollee(enrollee.getId()));
+        return loadEnrolleeDetails(enrollee);
+    }
+
+    /**
+     * Load enrollee tasks, profile, and kit requests
+     * (See loadForAdminView description for performance information)
+     */
+    public Enrollee loadEnrolleeDetails(Enrollee enrollee) {
```

Comment:
Sure. That name seems pretty specific for a general purpose layer, but are method names a way of identifying special use cases (like those that do joins)?
Also, it looks like we made a decision to do quasi joins at the DAO layer, rather than the service layer? Is that so we have the option to do actual joins at the DAO layer in the future?

Reply:
yeah, I think the theoretical best way to do these from a separation-of-concerns point of view is to make them `getEnrolleeWithRelations(id, relationNames)` so the caller can specify which joins they want.  But that obviously makes the methods more complex.  So yeah, explicitly calling out "this just loads the thing the PArticipant view needs" was my way of flagging that the separation of concerns wasn't clean.  

And yes, the quasi-joins in the DAO were for exactly the reason you say--we'll want to do them in SQL as soon as perf. becomes an issue (or maybe even before then, just as we realize the endpoints are relatively constant)